File: 3 КУРС/2 семестр/Java/JavaProects/app/src/main/java/com/example/voiceassistent/NumberToString.java
package com.example.voiceassistent;

import android.util.Log;


import androidx.core.util.Consumer;

import org.jetbrains.annotations.NotNull;

import java.util.Objects;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class NumberToString {
    public static void getNumber(String number, final Consumer<String> callback){
        NumberAPI api = NumberService.getApi();
        Call<Number> call = api.getNumber(number);
        call.enqueue(new Callback<Number>() {
            @Override
            public void onResponse(@NotNull Call<Number> call, @NotNull Response<Number> response) {
                Number result = response.body();
                if (result!=null) {
                    if (result.number!= null) {
                        String str=result.number;
                        StringBuilder newStr= new StringBuilder();
                        String []strr=str.split(" ");
                        for (int i=0; i<strr.length; i++)
                        {
                            if (!strr[i].contains("руб"))
                            {
                                newStr.append(strr[i]).append(" ");
                            }
                            else {
                                if (strr[i].contains("руб")) {
                                    i = str.length() - 1;
                                }
                            }
                        }
                        callback.accept(String.valueOf(newStr));
                    }
                    else
                        callback.accept("Не могу перевести result.str");
                }
                else callback.accept("Не могу перевести result");
            }
            @Override
            public void onFailure(@NotNull Call<Number> call, @NotNull Throwable t) {
                Log.w("NUMBER", Objects.requireNonNull(t.getMessage()));
                callback.accept("Не могу перевести failure");
            }
        });
    }
}

File: 3 КУРС/2 семестр/Учебная практика/asd/JavaProects/app/src/main/java/com/example/voiceassistent/NumberToString.kt
package com.example.voiceassistent

import android.util.Log
import java.util.function.Consumer
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import java.util.*

object NumberToString {
    fun getNumber(number: String?, callback: Consumer<String?>) {
        val api: NumberAPI = NumberService.api
        val call: Call<Number?>? = api.getNumber(number)
        call!!.enqueue(object : Callback<Number?> {
            override fun onResponse(call: Call<Number?>, response: Response<Number?>) {
                val result = response.body()
                if (result != null) {
                    if (result.number != null) {
                        val str = result.number
                        val newStr = StringBuilder()
                        val strr = str!!.split(" ").toTypedArray()
                        var i = 0
                        while (i < strr.size) {
                            if (!strr[i].contains("руб")) {
                                newStr.append(strr[i]).append(" ")
                            } else {
                                if (strr[i].contains("руб")) {
                                    i = str.length - 1
                                }
                            }
                            i++
                        }
                        callback.accept(newStr.toString())
                    } else callback.accept("Не могу перевести result.str")
                } else callback.accept("Не могу перевести result")
            }

            override fun onFailure(call: Call<Number?>, t: Throwable) {
                Log.w("NUMBER", Objects.requireNonNull(t.message))
                callback.accept("Не могу перевести failure")
            }
        })
    }
}